### ISSUE TITLE
Please add grub.cfg "i915.modeset=0" to boot Cherry-Trail devices

### ISSUE THREAD

**Issue by kapper1224** [hakuna-m/wubiuefi#27]
Hi.

Thank you for your "Wubi for Ubuntu".
There is no bug.

I have many ATOM Tablet device.
But Cherry-Trail Tablet (about UEFI 32bit devices) is blackscreen and no boot GUI when boot  on Ubuntu and any Linux distributions.
I think "Cherry-Trail devices” is needed "i915.modeset=0" in front of "quiet" in grub.cfg.
And I am playing Wubi for Ubuntu 16.10 on GPD-WIN now.
Please add "i915.modeset=0"  in grub.cfg.

Thank you!

references
http://linuxiumcomau.blogspot.com/2016/10/running-ubuntu-on-intel-bay-trail-and.html

Mytweet 
https://twitter.com/kapper1224/status/800267910333665281

**Comment by hakuna-m** [comment]
It is no problem to add some frequently used boot options to **/ubuntu/install/boot/grub/grub.cfg**. This grub.cfg is temporarily used during installation after first reboot. 

You can use this grub menu by pressing "ESC" if the following message appears:
```
Completing the Ubuntu installation."
For more installation boot options, press `ESC' now...

```
Currently there are 5 different menu entries:

- Normal mode `quiet splash --`
- Safe graphic mode `debug debug-ubiquity xforcevesa --`
- ACPI workarounds `debug debug-ubiquity -- acpi=off noapic nolapic`
- Verbose mode `debug debug-ubiquity --`
- Demo mode `quiet splash --` (boots into live environment of ISO)
 
My suggestion is that I will add
- Cherry Trail workarounds `i915.modeset=0 -- i915.modeset=0`

In general, I will try to update boot options. e.g. we should consider new usage of [before--after](https://help.ubuntu.com/community/BootOptions/before--after)

**Comment by kapper1224** [comment]
Thank you for your answer.

>My suggestion is that I will add
>Cherry Trail workarounds i915.modeset=0 -- i915.modeset=0
>In general, I will try to update boot options. e.g. we should consider new usage of before--after

I think there is no problem.
But after installed and reboot, there is need again, "i915.modeset=0"
Please care it.（change "update-grub"?)

Thank you.

**Comment by hakuna-m** [comment]
> But after installed and reboot, there is need again, "i915.modeset=0"

If I add it twice,  `i915.modeset=0` **after** ` -- ` should insert the boot option for the installed system in **/etc/default/grub**. But I will try it.

**Comment by kapper1224** [comment]
>If I add it twice, i915.modeset=0 after -- should insert the boot option for the installed system in /etc/default/grub. But I will try it.

I'm sorry it, I understand.
Thank you for your comment.

**Comment by hakuna-m** [comment]
I tried  `i915.modeset=0` before ` -- ` and also after ` -- ` . Unfortunately, it didn't add this option to **/etc/default/grub** . 

IMHO the reason is that the Ubuntu installer Ubiquity skips module specific options with dot. So it would work if we add options without dot like `nomodeset` but does `nomodeset` also work on Cherry Trail devices ?

If yes, I suggest to add a boot option `nomodeset` instead of `i915.modeset=0`.
if no, I try to remove the Ubiquity part which skips module specific options.

**Comment by kapper1224** [comment]
I tried "nomodeset" before "queit" in grub.cfg, Cherry-Trail devices also work no problem. 
But I noticed another problem, Cherry-Trail devices can boot 「force the Linux kernel to boot in normal text mode」 only.
Please delete ”gfxmode $linux_gfx_mode” in grub.cfg.
I test and boot Ubuntu it.

Thank you!

**Comment by hakuna-m** [comment]
> But I noticed another problem,

If I understand you well, this means that `nomodeset` causes other problems.

Meanwhile I patched file **/bin/user-params** of [package ubiquity](http://packages.ubuntu.com/ubiquity) for Wubiuefi. So it is now possible to add module specific options to **/etc/default/grub** during installation.

Even if not needed we are more flexibel for different boot options. I created a new version with this patch and new menu entries in the first GRUB menu; the GRUB menu which appears if you press `ESC` at
```
Completing the Ubuntu installation.
For more installation boot options, press `ESC' now... 
```
For you, I added `Intel graphics workarounds` with boot option `i915.modeset=0`. I tested it and the **/etc/default/grub** of the installed system contains
`GRUB_CMDLINE_LINUX="rootflags=sync i915.modeset=0"`
and the **/boot/grub/grub.cfg** contains e.g.
`linux	/boot/vmlinuz-4.8.0-22-generic root=UUID=ABCDEF0123456789 loop=/ubuntu/disks/root.disk ro rootflags=sync i915.modeset=0  quiet splash $vt_handoff` 
  
I uploaded the new version as [wubitest.exe](https://www.dropbox.com/sh/6uqomp8l1frcd1y/AACC3UyJjId3gkw6EiRZ6lGTa/wubitest.exe)

md5sum: 3ad07026baacd18e42a92876727beb22

If it also works for you, I will release it as wubi1610r318.exe

**Comment by kapper1224** [comment]
Thank you for your fix.
I tried and boot my GPD-WIN (Cherry-Trail) on "Intel graphics workarounds" options select only now.
There is no problem to boot it.

Thank you very much.

my twitter and screenshot
https://twitter.com/kapper1224/status/801203424536600576

**Comment by hakuna-m** [comment]
I set [1610r318](https://github.com/hakuna-m/wubiuefi/releases/tag/1610r318) as latest release because of positive feedbacks.

I also added `PAE mode`. It was the wish of some users of old Pentium M CPUs who had to add `forcepae` manually. It has nothing to do with the entry `Intel graphics workarounds` for Cherry Trail devices (and possible other devices with Intel graphics) except that it is also a boot option. 

IMHO it should be easy to change boot options for future needs. If there are other boot options which are frequently used by a lot of people, I will also add them as menu entry.
